### PR TITLE
Move health endpoint-related code into a separate file

### PR DESCRIPTION
**Pull request by palant**
## Description

This creates a new module `health` responsible for the health endpoint. So far it handles the server initialization in the `init()` method and request processing in the `pre_process()` method. Ideally, handling of the health setting in CLI and config file will also be added here in future (not trivial given how parsing is tied to data structures).

## Related Issue
This is a tiny first step towards #342. 

## How Has This Been Tested?
Tests pass, and I’ve added a few new tests as well. I did not perform any functional testing at this point.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -158,6 +158,7 @@ pub mod error_page;
 #[cfg_attr(docsrs, doc(cfg(feature = "fallback-page")))]
 pub mod fallback_page;
 pub mod handler;
+pub mod health;
```
Comment (joseluisq) [inline src/lib.rs]: No need to expose this module, just make it available within the crate.

File: src/handler.rs [hunk 2]
```diff
@@ -161,40 +187,25 @@ impl RequestHandler {
...
+            let mut uri_path = uri.path().to_owned();
+            let uri_query = uri.query();
+
+            // Health requests aren't logged here but in health module.
             tracing::info!(
                 "incoming request: method={} uri={}{}",
                 method,
                 uri,
                 remote_addr_str,
             );
```
Comment (joseluisq) [inline src/handler.rs:208]: We want to log only when the `health` feature is disabled. Can we add a conditional over here?

**Review by joseluisq — CHANGES_REQUESTED** [review]
Just minor changes but looks fine to me overall.

File: src/lib.rs (hunk 1 above)
Comment (palant) [reply src/lib.rs]: Sure, I wasn’t certain what the policy on exposing APIs was – currently it looks like everything is being exposed.

File: src/handler.rs (hunk 2 above)
Comment (palant) [reply src/handler.rs:208]: I’m not sure, I think one of us misunderstands this. There is a conditional already, a few lines above. If `health` feature is enabled, `health::pre_process()` call will produce a result, so we go into `return result` and don’t get to this logging code. This should be good enough already?

Comment (joseluisq) [reply src/handler.rs:208]: You're right. I just missed the return.

**Review by joseluisq — APPROVED** [review]
Thanks!

**Pull request merged**